bin/src/command/orders.rs

```diff
@@ -127,70 +114,56 @@ impl CommandServer {
 
         info!("wrote {} commands to {}", counter, path);
 
-        Ok(OrderSuccess::SaveState(counter, path.into()))
+        Ok(Some(Success::SaveState(counter, path.into())))
     }
 
-    pub fn save_state_to_file(&mut self, f: &mut fs::File) -> anyhow::Result<usize> {
+    pub fn save_state_to_file(&mut self, file: &mut File) -> anyhow::Result<usize> {
         let mut counter = 0usize;
         let orders = self.state.generate_orders();
 
-        let res: io::Result<usize> = (move || {
+        let result: io::Result<usize> = (move || {
             for command in orders {
                 let message = CommandRequest::new(
                     format!("SAVE-{}", counter),
                     CommandRequestData::Proxy(command),
                     None,
                 );
 
-                f.write_all(
+                file.write_all(
                     &serde_json::to_string(&message)
                         .map(|s| s.into_bytes())
                         .unwrap_or(vec![]),
                 )?;
-                f.write_all(&b"\n\0"[..])?;
+                file.write_all(&b"\n\0"[..])?;
 
                 if counter % 1000 == 0 {
                     info!("writing command {}", counter);
-                    f.sync_all()?;
+                    file.sync_all()?;
                 }
                 counter += 1;
             }
-            f.sync_all()?;
+            file.sync_all()?;
 
             Ok(counter)
         })();
 
-        Ok(res?)
+        Ok(result?)
```

Comment:
Could you add some context using `.with_context(...)`

Reply:
sure!